File: PortfolioTracker.Api/Dtos/Transaction.cs
public class Transaction
{
    public int Id {get;set;}
    public int UserId {get;set;}
    public int AssetId {get;set;}
    public TransactionType Type {get;set;}
    public decimal Quantity {get;set;}
    public decimal Price {get;set;}
    public decimal TotalAmount {get;set;}
    public DateTime Date {get;set;}
    public decimal Fees {get;set;}
    public string Notes {get;set;} = string.Empty;
    
}

File: my-app/src/types/Transaction.ts
export interface Transaction {
  Id: number;
  UserId: number;
  AssetId: number;
  
  AssetTicker: string;
  AssetName: string;
  AssetIcon: string;
  Type: 'BUY' | 'SELL'; 
  
  Quantity: number;
  Price: number;
  TotalAmount: number;
  Date: string;
  Fees: number;
  Notes: string;
}